src/main/java/org/primefaces/extensions/validate/PhoneValidator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.validate;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.validator.FacesValidator;
+import javax.faces.validator.Validator;
+import javax.faces.validator.ValidatorException;
+import org.primefaces.extensions.component.inputphone.InputPhone;
+
+/**
+ * Validator used with {@link InputPhone}.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ */
+@FacesValidator("org.primefaces.extensions.validate.PhoneValidator")
+public class PhoneValidator implements Validator {
```

Comment:
I'm not sure if we want to validate by default, so I opted to create a validator class which can be used optionally. We could move the code to the renderer and validate based on an attribute flag.

Reply:
I would like @tandraschko or @cnsgithub to weigh in but typically we don't make this optional we enforce validation in the rendering on Submit.

Reply:
Basically, opt-in input validation means opt-in security as well, which isn't a good idea, is it?! :wink:

Reply:
However, just keep in mind, that implementations on client and server side may differ and result in unexpected behaviour, i.e. client side validation passes but server side validation does not and vice versa.

Reply:
> Basically, opt-in input validation means opt-in security as well, which isn't a good idea, is it?!

It might depend on what you consider valid. For example: "1-800-pizza", is that a valid phone number? This validator should be a reasonable default I would say.

Reply:
There might of course be use-case-dependent validation rules. Nevertheless, if a phone number gets rejected on client side, it HAS TO BE rejected on server side as well. Therefore, it's essential for the client and server side implementations to be symmetric.

Reply:
I don't think any checks are done on the client side. I'll see if we can have the server side to be as lenient as possible.

Reply:
Okay, I have added the validator by default.